### PR TITLE
Bug fix on JdkHelper + tests.

### PR DESCRIPTION
**Pull request by chenlevy24**
Fixed a bug on 'JdkHelper's' 'validatePath' function and added tests to it.

### CODE REVIEW

**Comment by chenlevy24** [comment]
I didn't test the static method of `getJavaHome` because it just calls the original one, there is nothing new to test

**Comment by andreban** [comment]
> I didn't test the static method of `getJavaHome` because it just calls the original one, there is nothing new to test

Isn't it the other way around? the `getJavaHome()` that's **not** static calls the static one using instance variables from `JdkHelper`.

**Comment by chenlevy24** [comment]
Right, I got confused but I know how it works.. anyway the bottom line is the same, I don't think that a new test for this function is needed

**Comment by andreban** [comment]
`getJavaHome()` itself doesn't seem to have any tests. `getEnv()` does have tests, and currently tests it indirectly. I don't think `getJavaHome()` was being used elsewhere (it should probably have been marked as private). We shouldn't depend on the `getEnv()` test to check it, as the implementation there can change. I'm happy with adding a test for the static version and skipping the instance version, but we should test one of the two.

File: packages/core/src/spec/lib/jdk/JdkHelperSpec.ts [hunk 1]
```diff
@@ -63,4 +64,25 @@ describe('JdkHelper', () => {
       expect(env['JAVA_HOME']).toBe('C:\\Users\\user\\jdk8\\');
     });
   });
+
+  describe('validatePath', () => {
+    it('Checks that given a valid path, validatePath will pass', async () => {
```
Comment (andreban) [inline packages/core/src/spec/lib/jdk/JdkHelperSpec.ts:69]: Can you add tests for the path validation each platform, similar to what the `getEnv()` test does?

**Review by andreban — CHANGES_REQUESTED** [review]

**Review by PEConn — APPROVED** [review]
Code LGTM once Andre approves.

File: packages/core/src/spec/lib/jdk/JdkHelperSpec.ts (hunk 1 above)
Comment (chenlevy24) [reply packages/core/src/spec/lib/jdk/JdkHelperSpec.ts:69]: I can, but if `getJavaHome()` it proven to be working those tests will mean nothing.. `validatePath` uses `getJavaHome` and after we got to this folder it all looks the same

Comment (andreban) [reply packages/core/src/spec/lib/jdk/JdkHelperSpec.ts:69]: Sounds good, adding tests to `getJavaHome()` should be enough.

File: packages/core/src/spec/lib/jdk/JdkHelperSpec.ts [hunk 2]
```diff
@@ -63,4 +64,86 @@ describe('JdkHelper', () => {
...
+    it('Creates a Windows environment and checks that a valid path will pass', async () => {
+      mock({
+        'jdk': {
+          'release': 'JAVA_VERSION="1.8',
+        }});
+      expect((await JdkHelper.validatePath('jdk')).isOk()).toBeTrue();
+      mock.restore();
+    });
+
+    it('Creates a Windows environment and checks that a valid path will not pass', async () => {
```
Comment (PEConn) [inline packages/core/src/spec/lib/jdk/JdkHelperSpec.ts:78]: "checks that a valid path will not pass" ->
"checks that an invalid path will not pass"

**Review by PEConn — APPROVED** [review]

**Review by andreban — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**